contracts/pcv/curve/StableSwapOperatorV1.sol

```diff
@@ -0,0 +1,264 @@
+            IERC20(_3crv).approve(pool, 0);
+            IERC20(_3crv).approve(pool, _3crvBalanceAfter);
+
+            // do deposit
+            uint256 _balanceBefore = IERC20(pool).balanceOf(address(this));
+            IStableSwap(pool).add_liquidity(_addLiquidityAmounts, 0);
+            uint256 _balanceAfter = IERC20(pool).balanceOf(address(this));
+
+            // emit event
+            uint256 _lpTotalSupply = IERC20(pool).totalSupply();
+            uint256 _3crvOut = (_3crvAmountAfter * (_balanceAfter - _balanceBefore)) / _lpTotalSupply;
+            uint256 _daiOut = IStableSwap(_3pool).calc_withdraw_one_coin(
+              _3crvOut, // LP tokens just deposited
+              0 // 3pool coin 0 = DAI
+            );
+            emit Deposit(msg.sender, _daiOut);
+        }
+
+        // burn remaining FEI
+        uint256 _feiBalance = _fei.balanceOf(address(this));
+        _fei.burn(_feiBalance);
+    }
+
+    /// @notice withdraw DAI from the deposit. This will remove liquidity in the
+    /// current pool's proportion, burn the FEI part of it, and then use the 3crv
+    /// to withdraw liquidity of the 3pool in DAI.
+    /// note: because of slippage & fees on withdraw, the amount of DAI out is
+    ///        not exactly the expected amountUnderlying provided as input.
+    /// @param amountUnderlying of tokens withdrawn (0 = all)
+    /// @param to the address to send PCV to (self = no transfer)
+    function withdraw(address to, uint256 amountUnderlying)
+        public
+        override
+        onlyPCVController
+        whenNotPaused
+    {
+
+        uint256 _3crvVirtualPrice = IStableSwap(_3pool).get_virtual_price();
+        // withdraw all
+        if (amountUnderlying == 0) {
+            uint256 _lpToWithdraw = IERC20(pool).balanceOf(address(this));
+            uint256[2] memory _minAmounts; // [0, 0]
+            IERC20(pool).approve(pool, _lpToWithdraw);
+            IStableSwap(pool).remove_liquidity(_lpToWithdraw, _minAmounts);
+        }
+        // partial withdraw
+        else {
+            uint256 _amount3crvToWithdraw = amountUnderlying * 1e18 / _3crvVirtualPrice;
+            uint256 _lpTotalSupply = IERC20(pool).totalSupply();
+            uint256 _lpToWithdraw = _lpTotalSupply * _amount3crvToWithdraw / IStableSwap(pool).balances(_3crvIndex);
+
+            uint256[2] memory _minAmounts; // [0, 0]
+            IERC20(pool).approve(pool, _lpToWithdraw);
+            IStableSwap(pool).remove_liquidity(_lpToWithdraw, _minAmounts);
+        }
+
+        // remove 3pool liquidity in DAI (3crv -> DAI)
+        uint256 _3crvBalance = IERC20(_3crv).balanceOf(address(this));
+        uint256 _daiBalanceBefore = IERC20(_dai).balanceOf(address(this));
+        IERC20(_3pool).approve(_3pool, _3crvBalance);
+        IStableSwap(_3pool).remove_liquidity_one_coin(
+          _3crvBalance,
+          0, // 0 = DAI token index
+          0 // minimum 0 DAI out (defensively checked below)
```

Comment:
same comment as above, if we can use the embedded slippage check that is preferrable